lib/stack_master/template_compiler.rb

```diff
@@ -2,12 +2,12 @@ module StackMaster
   class TemplateCompiler
     TemplateCompilationFailed = Class.new(RuntimeError)
 
-    def self.compile(config, template_file_path, compile_time_parameters, compiler_options = {})
-      compiler = template_compiler_for_file(template_file_path, config)
+    def self.compile(config, template_dir, template_file_path, sparkle_pack_template, compile_time_parameters, compiler_options = {})
```

Comment:
Having a sparkle specific option on the generic TemplateCompiler isn't ideal... I think it would be better as a compiler_option, or even as it was before using the `template` key and a compiler option.

Reply:
@patrobinson Any opinions? As a user I would expect the template name in the stack options (which I understand as "what to compile") as opposed to compiler option (which I understand as "how to compile"), but I can change my mind :)
I'm still happy to move it around if there's an agreement on a better solution.

Reply:
@stevehodgkiss I'm confused, I thought you were in support of this change https://github.com/envato/stack_master/pull/286#discussion_r313808613

Reply:
That comment didn't convey what I intended clearly enough. In my head all the `sparkle_` options would have been compiler options. I don't think compiler specific options should be added directly to the stack definition.

StackMaster supports multiple template compilers - json/yaml/cfndsl/sparkleformation… if we added all their options as arguments to the template compiler (via the stack definition) it would be a mess.

Going forward from here, I'd propose either as it was in this PR before:

```
my-stack:
  template: sparkle_template.rb
  compiler_options:
    sparkle_packs:
      - some-sparkle-pack
    sparkle_pack_template: true
```

Or

```
my-stack:
  compiler_options:
    sparkle_packs:
      - some-sparkle-pack
    sparkle_pack_template: sparkle_template.rb
```

Thinking about it some more, I prefer the first one...

Reply:
FWIW, I'm a +1 on the first option.

Reply:
@stevehodgkiss Your argument makes sense to me. Sounds like a good separation of concerns.

I don't know much about the inner-workings of Stack Master to understand the consequences, but from looking at the stack snippet options, from a user's perspective, I like option 1.

Reply:
The first option was how we had it originally. I wasn't keen on the extra `sparkle_pack_template: true` option because it was a bit superfluous IMO given you could just supply a different parameter but I'm happy either way.

Reply:
Pushed a PR with those changes and some testing improvements https://github.com/envato/stack_master/pull/289